src/NzbDrone.Core/Applications/Radarr/RadarrV3Proxy.cs

```diff
@@ -102,7 +102,17 @@ public ValidationFailure TestConnection(RadarrIndexer indexer, RadarrSettings se
 
             try
             {
-                Execute<RadarrIndexer>(request);
+                var version = _httpClient.Post<RadarrIndexer>(request).Headers.GetSingleValue("X-Application-Version");
+
+                if (version == null)
+                {
+                    return new ValidationFailure(string.Empty, "Failed to fetch Radarr version");
+                }
+
+                if (new Version(version) < new Version("4.0.0.0"))
```

Comment:
first v4 stable is `[4.0.4.5922](https://github.com/Radarr/Radarr/releases/tag/v4.0.4.5922)`